third_party/ienn/src/ie_compilation.cpp

```diff
@@ -170,8 +170,15 @@ int32_t Compilation::AddInput(uint32_t index) {
     return result;
   }
   try {
-    auto input_node =
-        std::make_shared<op::Parameter>(element::f32, Shape(dims));
+    bool fp32_precision = preference_ != PREFER_LOW_POWER;
+    std::shared_ptr<op::Parameter> input_node;
+    if (fp32_precision) {
```

Comment:
`auto input_node =	std::make_shared<op::Parameter>(preference_ != PREFER_LOW_POWER ? element::f32 : element::f16, Shape(dims));	`

Reply:
Ok, thanks! If change as above, need I add comments to explain the f32/f16?

Reply:
> Ok, thanks! If change as above, need I add comments to explain the f32/f16?

There are some comments in `addConstant()` already, still need to add here?

Reply:
that's fine, could you please add some explain for the reason of using f16 but the input precision is set f32.

Reply:
OK，I will.